### ISSUE TITLE
[ACR] The token returned by  `az acr login --expose-token`  is a refresh token instead of an access token

### ISSUE THREAD

**Issue by Wwwsylvia** [Azure/azure-cli#20689]
## Describe the bug

**Command Name**
`az acr login`

**Errors:**
![image](https://user-images.githubusercontent.com/8113721/145549014-8fdc8820-6cb3-4349-b5df-ba2c00e2548a.png)

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az acr login -n {} --expose-token`

## Expected Behavior
The token returned by  `az acr login --expose-token`  is a refresh token instead of an access token. The name `accessToken` is misleading.
We should correct this, but there will be a breaking change.
See https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py#L362-L369 and https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py#L149-L151
Reference: https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#calling-post-oauth2exchange-to-get-an-acr-refresh-token

## Environment Summary
```
Windows-10-10.0.22000-SP0
Python 3.8.9
Installer: MSI

azure-cli 2.31.0

```

**Comment by jiasli** [comment]
Routing to service team.

**Comment by toddysm** [comment]
@johnsonshi Can you please check if this is still an issue or should we close this in 30 days?

**Comment by Wwwsylvia** [comment]
@toddysm I think the current behavior might be confusing to users. Do we consider outputting something like below for clarity? We might still need to keep the `accessToken` field for a while for compatibility though.

```json
{
  "username": "00000000-0000-0000-0000-000000000000",
  "accessToken": "the_token",
  "refereshToken": "the_token",
  "loginServer": "lixleitest0.azurecr.io"
}
```

**Comment by johnsonshi** [comment]
Yes indeed, the token exposed from `az acr login --expose-token` is an ACR Refresh Token. You will use this ACR Refresh Token whenever you perform an OAuth authenticated action on the ACR data plane, at which point you will exchange an ACR Refresh Token for an ACR Access Token to perform that specific data plane operation.

The return value from `az acr login --expose-token` is indeed confusing as the relevant field is `accessToken` instead of `refreshToken`. Changing it would break a lot of downstream customers, so @Wwwsylvia's suggestion on having an extra field for `refreshToken` is the way to go.

**Comment by Wwwsylvia** [comment]
@m5i-work My concern is that the name `accessToken` is not consistent with the term `refereshToken` we use in this public doc [Authenticating docker with an ACR refresh token](https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#authenticating-docker-with-an-acr-refresh-token) and this might be misleading to users.

BTW, I think would be more user-friendly to add a `username` field in the result, as the username hint in the message 
> You can perform manual login using the provided access token below, for example: 'docker login loginServer -u 00000000-0000-0000-0000-000000000000 -p accessToken'

 is not obvious and users might think `00000000-0000-0000-0000-000000000000` is just a placeholder...

**Comment by wangxiaoxuan273** [comment]
I'm working on this issue now. I tried implementation and the fields are shown in alphabetical order by the cli:
```
{
  "accessToken": "the_token",
  "loginServer": "myregistry.azurecr.io",
  "refreshToken": "the_token",
  "username": "00000000-0000-0000-0000-000000000000"
}
```

**Comment by johnsonshi** [comment]
I disagree with @m5i-work. Not all users will be reading documentation.
* The ones that read documentation will know that the `accessToken` field in the `az acr login --expose-token` output pertains to an ACR Refresh Token that was obtained from exchanging an Entra AAD Token.
  * And if they are familiar with Docker terminology, like mentioned by others earlier, they will know that this field is called `accessToken` due to Docker standards set earlier.
* The ones that do not read documentation will be confused by the `accessToken` field and will think that this `accessToken` can be used directly during the image push/pull flow.
  * By having `refreshToken` in the output, it clearly communicates to them that they can use this `refreshToken` output as an ACR Refresh Token in later workflows that require changing an ACR Refresh Token for an ACR Access Token.

So I agree with @Wwwsylvia and @wangxiaoxuan273 with the following approach, except that there should be an additional diagnostic message outputted to `stderr`:

```json
Note: The token in both the accessToken and refreshToken fields is an ACR Refresh Token, not an ACR Access Token. This ACR Refresh Token must be used to obtain an ACR Access Token when performing data plane operations, such as pushing or pulling images.
{
  "accessToken": "the_token",
  "loginServer": "myregistry.azurecr.io",
  "refreshToken": "the_token",
  "username": "00000000-0000-0000-0000-000000000000"
}
```

**Comment by wangxiaoxuan273** [comment]
If we have the additional diagnostic message, I don't think we need to add the extra `refreshToken` field, it may confuse the users. We can only have the following fields and have the diagnostic message as "The token in the accessToken field is an ACR Refresh Token, not an ACR Access Token......."
```
{
  "accessToken": "the_token",
  "loginServer": "myregistry.azurecr.io",
  "username": "00000000-0000-0000-0000-000000000000"
}
```

**Comment by Wwwsylvia** [comment]
> I disagree with [@m5i-work](https://github.com/m5i-work). Not all users will be reading documentation.
> 
> * The ones that read documentation will know that the `accessToken` field in the `az acr login --expose-token` output pertains to an ACR Refresh Token that was obtained from exchanging an Entra AAD Token.
>   
>   * And if they are familiar with Docker terminology, like mentioned by others earlier, they will know that this field is called `accessToken` due to Docker standards set earlier.
> * The ones that do not read documentation will be confused by the `accessToken` field and will think that this `accessToken` can be used directly during the image push/pull flow.
>   
>   * By having `refreshToken` in the output, it clearly communicates to them that they can use this `refreshToken` output as an ACR Refresh Token in later workflows that require changing an ACR Refresh Token for an ACR Access Token.
> 
> So I agree with [@Wwwsylvia](https://github.com/Wwwsylvia) and [@wangxiaoxuan273](https://github.com/wangxiaoxuan273) with the following approach, except that there should be an additional diagnostic message outputted to `stderr`:
> 
> Note: The token in both the accessToken and refreshToken fields is an ACR Refresh Token, not an ACR Access Token. This ACR Refresh Token must be used to obtain an ACR Access Token when performing data plane operations, such as pushing or pulling images.
> {
>   "accessToken": "the_token",
>   "loginServer": "myregistry.azurecr.io",
>   "refreshToken": "the_token",
>   "username": "00000000-0000-0000-0000-000000000000"
> }

LGTM
